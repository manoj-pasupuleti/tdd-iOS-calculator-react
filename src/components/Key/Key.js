import React from "react";
import PropTypes from "prop-types";

import "./Key.css";

export default function key({ handleKeyDown, keyAction, keyType, keyValue }) {
  const keyClass = `key-container ${keyType}`;
  return (
    <div
      className={keyClass}
      onClick={() => keyAction(keyValue)}
      onKeyPress={handleKeyDown}
      role="button"
      tabIndex="0"
      aria-pressed="false"
      id={`${keyType}-${keyValue}`}
    >
      <p className="key-value">{keyValue}</p>
    </div>
  );
}

key.propTypes = {
  handleKeyDown: PropTypes.func.isRequired,
  keyAction: PropTypes.func.isRequired,
  keyType: PropTypes.string.isRequired,
  keyValue: PropTypes.string.isRequired
};
