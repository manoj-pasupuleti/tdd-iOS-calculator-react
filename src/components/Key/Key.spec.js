import React from "react";
import { shallow } from "enzyme";
import sinon from "sinon";
import Key from "./Key";

describe("Key", () => {
  let wrapper;
  const onClick = sinon.spy();
  beforeEach(() => {
    wrapper = shallow(
      <Key
        handleKeyDown={jest.fn()}
        keyAction={onClick}
        keyType="number-key"
        keyValue="0"
      />
    );
  });

  it("Should Render Correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Should Render key Value Correctly", () => {
    expect(wrapper.text()).toEqual("0");
  });

  it("Simulate click action", () => {
    wrapper.find("div").simulate("click");
    expect(onClick.callCount).toEqual(1);
  });
});
