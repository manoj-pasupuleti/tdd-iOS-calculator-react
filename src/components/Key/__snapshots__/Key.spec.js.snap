// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Key Should Render Correctly 1`] = `
<div
  aria-pressed="false"
  className="key-container number-key"
  id="number-key-0"
  onClick={[Function]}
  onKeyPress={[MockFunction]}
  role="button"
  tabIndex="0"
>
  <p
    className="key-value"
  >
    0
  </p>
</div>
`;
